src/main/java/com/gojek/de/stencil/DescriptorMapBuilder.java

```diff
@@ -27,26 +27,28 @@
         fileDescriptors.forEach(fd -> {
             String javaPackage = fd.getOptions().getJavaPackage();
             String protoPackage = fd.getPackage();
-            fd.getMessageTypes().stream().forEach(desc -> {
-                String className = desc.getName();
-                desc.getNestedTypes().stream().forEach(nestedDesc -> {
-                    String nestedClassName = nestedDesc.getName();
-                    descriptorMap.put(
-                            String.format("%s.%s.%s", javaPackage, className, nestedClassName),
-                            new DescriptorAndTypeName(
-                                    nestedDesc,
-                                    String.format(".%s.%s.%s", protoPackage, className, nestedClassName)
-                            ));
-                });
-                descriptorMap.put(
-                        String.format("%s.%s", javaPackage, className),
-                        new DescriptorAndTypeName(
-                                desc,
-                                String.format(".%s.%s", protoPackage, className)
-                        ));
-            });
+            fd.getMessageTypes().stream().forEach(desc -> descriptorMap.putAll(getFlattenedDescriptors(desc, javaPackage, protoPackage, "", new HashMap<>())));
         });
 
         return descriptorMap;
     }
+
+    private Map<String, DescriptorAndTypeName> getFlattenedDescriptors(Descriptors.Descriptor descriptor, String javaPackage, String protoPackage, String parentClassName, Map<String, DescriptorAndTypeName> initialDescriptorMap) {
+        String className = getClassName(descriptor, parentClassName);
+        initialDescriptorMap.put(
+                String.format("%s.%s", javaPackage, className),
+                new DescriptorAndTypeName(
+                        descriptor,
+                        String.format(".%s.%s", protoPackage, className)
+                ));
+        descriptor.getNestedTypes()
```

Comment:
@sravankorumilli can you please add a test for recursive type protobuf? I think the current code will break for recursive one

Reply:
code changes LGTM

Reply:
This works well @sravankorumilli I have verified by running the test locally in debug mode